### ISSUE TITLE
Add support for parsing HTML / literal text

### ISSUE THREAD

**Issue by ryangjchandler** [tuskphp/parser#3]
A normal PHP parser would be able to parse out HTML & other non-PHP related code (outside of a `<?php` tag). At the moment, we just assume everything is inside of a `<?php` tag which is problematic.

I think the smartest way to resolve this is by merging the lexer and parser into a single crate, then migrating to a state machine pattern for the lexer itself.

That way, we can have a 2-state lexer (non-php & php).

**Comment by rlcurrall** [comment]
Do you intend to merge the two repositories (**tuskphp/lexer** and **tuskphp/parser**) into this repository?

**Comment by ryangjchandler** [comment]
@rlcurrall Yeah, I'm planning on creating a single repository for the Tusk core using workspaces. That'll contain the lexer/parser in a single crate, the engine in another crate & also the standard library in another.
